test/smoke/generated/msi-resource-manager/src/models/mappers.ts

```diff
@@ -38,22 +38,12 @@ export const Resource: coreHttp.CompositeMapper = {
   }
 };
 
-export const ProxyResource: coreHttp.CompositeMapper = {
-  type: {
-    name: "Composite",
-    className: "ProxyResource",
-    modelProperties: {
-      ...Resource.type.modelProperties
-    }
-  }
-};
-
 export const SystemAssignedIdentity: coreHttp.CompositeMapper = {
   type: {
     name: "Composite",
     className: "SystemAssignedIdentity",
     modelProperties: {
-      ...ProxyResource.type.modelProperties,
+      ...Resource.type.modelProperties,
```

Comment:
@daviwil this change seems to be caused by the last version of M4. I thought the step to remove these empty parents was skipped for now. Do I need to update something in my autorest command?

This is the definition of ProxyResource ([link](https://github.com/Azure/azure-rest-api-specs/blob/e994a9b58f099f073dd60bba3fd53d7bd6e969a6/specification/common-types/resource-management/v1/types.json#L78))

```json
 "ProxyResource": {
      "description": "The resource model definition for a ARM proxy resource. It will have everything other than required location and tags",
      "allOf": [
        {
          "$ref": "#/definitions/Resource"
        }
      ]
    }
```

Reply:
@joheredi 
Do you mean that the `ProxyResource` mapper shouldn't be generated, but it is because it's in the code model?

Reply:
Originally I thought that ProxyResource shouldn't have been removed. But it looks like I was wrong 😄. I think M4 removing it is expected with the last version.

Some background on why I thought ProxyResource shouldn't have been removed, feel free to ignore:

So there was some logic added in M4 recently that removes parents if the only thing they do is extend from another object.

In a recent PR, David mentioned that this logic was going to be disabled ([link](https://github.com/Azure/autorest.modelerfour/pull/274#issuecomment-624300730)), so I thought the removal of ProxyResource might've been unintended.

However I clearly didn't reach the bottom of the PR,  where David said this logic would be partially enabled instead of completely disabled ([link](https://github.com/Azure/autorest.modelerfour/pull/274#issuecomment-624329254)).